### ISSUE TITLE
Google Analytics still tracks on GitHub - collection sent via POST

### ISSUE THREAD

**Issue by mr-bombastiq** [gorhill/uBlock#870]
![screen shot 2015-10-27 at 5 01 03 pm](https://cloud.githubusercontent.com/assets/15337293/10750781/cc6e7f00-7ccd-11e5-8842-a5ce43fe960f.png)

Google Analytics is still able to receive requests. You can see this on GitHub.com itself.

I am guessing this is due to:

1. Google Analytics being minified into one file by GitHub (the Google Analytics is included in the one ".js" file that is downloaded).
2. The request is a POST request, not a GET request.

**Comment by mr-bombastiq** [comment]
It is important to note that this particular POST request for Google Analytics *does not* appear in the logger.

**Comment by harshanvn** [comment]
i don't see this request either in uBlock (v1.3.1 or v1.3.2rc4). Tested in FF v41.0.2/
See the screenshot below -
which browser are you using? Maybe another extension doing it, try disabling all your extensions and see?

![ublock 870](https://cloud.githubusercontent.com/assets/779630/10752101/93f5c1cc-7c5c-11e5-8889-fbbfc8f2124b.png)

**Comment by seanrand** [comment]
Github.com is using navigator.sendBeacon which will show up in the behind-the-scene tab of the logger.

**Comment by mr-bombastiq** [comment]
@harshanvn I was using Chrome.

@seanrand, yes, you are correct. Adding the following rule seems to stop the requests

```
||www.google-analytics.com/collect$object,domain=behind-the-scene
```

**Comment by seanrand** [comment]
The "Disable hyperlink auditing/beacon" option and the linked wilderssecurity thread in the uBlock settings appears to be a bit misleading to me, as the option doesn't affect sendBeacon but only the hyperlink auditing via ping attribute AFAIK. (At least in Chromium?)

**Comment by RoxKilly** [comment]
Web beacons allow websites to send data to a server in the background usually as you navigate away from the page. These requests are not part of the traditional back and forth that occur as a browser loads a webpage. 

In Firefox, you can disable them by navigating to `about:config` and  setting `beacon.enabled=false`

**Comment by gorhill** [comment]
> It is important to note that this particular POST request for Google Analytics does not appear in the logger.

It does, it appears in the behind-the-scene context. Select "All" in the logger.

**Comment by gorhill** [comment]
> The "Disable hyperlink auditing/beacon" option and the linked wilderssecurity thread in the uBlock settings appears to be a bit misleading to me

Yes, I should write my own wiki page with caveats, and link to this page as further resource.

There is no setting to disable `sendBeacon` on Chromium, unlike Firefox, and unfortunately network requests related to `sendBacon` do not have some specific content type that uBO could intercept and block according to whether hyperlink/beacon is checked.

**Comment by tntc4** [comment]
I have a GreaseMonkey (FF) script writen by myself to remove navigator.sendBeacon functionality of FF.
https://openuserjs.org/install/tntc4/navigator.sendBeacon_Remover.user.js

**Comment by RoxKilly** [comment]
@tntc4 Can your script be used effectively in Chrome?  Firefox already has a config option to disable beacons(set `beacon.enabled=false`), but I don't think Chrome has such an option.

Your script comments state:
> To-do: send ajax request and report url wich using navigator.sendBeacon usage and params

In my opinion, this would be counterproductive at best, and deceive users at worst. Users try to turn off beacons to enhance their privacy, but sending all beacon data from all sites to one server would damage privacy I think.

**Comment by tntc4** [comment]
yesterday I saw this issue and just release my script for publick usage.
I just want to help people stop and block data leading by beacons.
I was plan to implement this function in script that I'm only user that uses - to track websites that uses beacons (for my purposes).
I'm never release such option in public scripts.
I will update script description to avoid misundestanding.
thank you.

**Comment by BenjaminProgram** [comment]
@gorhill I haven't read through any of the documentation or code since before Origin. I finally decided to update. Would it work to block all google-analytics requests?

**Comment by BenjaminProgram** [comment]
@tntc4 Would you implement that into uBlock Origin?

**Comment by tntc4** [comment]
@BenjaminProgram I made a few modifications of my Grease Monkey script and test is localy but without success.
Still can't bypass Cross Origin Requests with Grease Monkey ajax engine for FireFox.
I'm not a very skilled javascript programer... sorry.

**Comment by BenjaminProgram** [comment]
@tntc4 I am not sure if any of this would be useful but below is what I turned up:

http://stackoverflow.com/questions/3076414/ways-to-circumvent-the-same-origin-policy

https://jvaneyck.wordpress.com/2014/01/07/cross-domain-requests-in-javascript/

**Comment by gorhill** [comment]
> Would it work to block all google-analytics requests?

uBO already blocks all Google Analytics requests. As said [above](https://github.com/gorhill/uBlock/issues/870#issuecomment-151484185), the issue here is that the network requests sent through `navigator.sendBeacon` are sent as behind-the-scene network requests in **Chromium**. However `behind-the-scene` scope is whitelisted by default, and only in advanced-user mode removing the whitelisting of the `behind-the-scene` is going to have any effects, due to the potentially negative consequences of doing so -- like crippling the browser in bad ways. I personally un-whitelisted the behind-the-scene scope -- so those pings are blocked on my side -- but then I do understand the consequences and won't open an issue here if ever something is unexpectedly blocked.

> @tntc4 Would you implement that into uBlock Origin?

I don't accept pull requests.

All of this may become a non-issue if even the fix to [this Chromium issue](https://code.google.com/p/chromium/issues/detail?id=522124) lands, and if the fix gives beacon network requests their own request type in the webRequest API.

**Comment by RoxKilly** [comment]
@gorhill what do you think about setting the Firefox config option `beacon.enabled=false` to disable beacons for Firefox users who have checked "Disable hyperlink auditing/beacon" in uBO's settings?

**Comment by gwarser** [comment]
> It's implemented

**Comment by RoxKilly** [comment]
Thanks
